linalg/hypre.hpp

```diff
@@ -989,9 +1034,83 @@ class HypreBoomerAMG : public HypreSolver
        As with SetSystemsOptions(), this solver assumes Ordering::byVDIM. */
    void SetElasticityOptions(ParFiniteElementSpace *fespace);
 
+#if MFEM_HYPRE_VERSION >= 21800
+   /* distance parameter takes on values {1,2,15} for lAIR, meaning R is built using
+   distance 1 neighbors, distance two neighbors, or distance two on processor and
+   distance 1 off processor (i.e., distance 1.5 --> 15).        */
+   void SetLAIROptions(int distance=15,  std::string prerelax="",
+                       std::string postrelax="FFC", double strength_tol=0.1,
+                       double strength_tolR=0.01, double filter_tolR=0.0,
+                       int interp_type=100, int relax_type=3, double filterA_tol=0.0,
+                       int splitting=6, int blksize=0, int Sabs=0);
+
+   void SetNAIROptions(int neumann_degree=2,  std::string prerelax="A",
+                       std::string postrelax="F", double strength_tol=0.1,
+                       double strength_tolR=0.01, double filter_tolR=0.0,
+                       int interp_type=100, int relax_type=10, double filterA_tol=0.0,
+                       int splitting=6, int blksize=0, int Sabs=0);
```

Comment:
Might want to reformat to keep these lines < 80 characters.

Reply:
Also, I think it's better to use `const char *` here instead of `std::string`. See my other comments in `hypre.cpp` and `ex9p.cpp` for related suggestions.

Reply:
Why const  char * as oppose to string?

Reply:
It doesn't make a big difference, but none of the features of `std::string` are used in the function, so `char *` is perhaps more "direct".